packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -65,10 +74,25 @@ export class AmountSelector extends BaseComponent {
     }
   };
 
+  requestUpdate(
+    name?: PropertyKey,
+    oldValue?: unknown,
+    options?: PropertyDeclaration<unknown, unknown>
+  ): void {
+    super.requestUpdate(name, oldValue, options);
+    if (name === BALANCE_UPDATE_KEY) {
+      this._validateAmount(String(this.amount));
+    }
+  }
+
   _onResourceSelectedHandler = ({ value }: DropdownOption<Resource>): void => {
     this.selectedResource = value;
     this.amount = 0;
-    this.tokenBalanceController.startBalanceUpdates(value);
+
+    this.tokenBalanceController.startBalanceUpdates(
+      this.selectedResource,
+      this.sourceSubstrateProvider
```

Comment:
and pass sourceDomainId here.
Btw have you tried using widget? I'm getting unsupported resource for rococco-phala -> Sepolia because this is undefined. Should we hardcode some defaults?

Reply:
@mpetrunic so substrate providers need to be provided by the consumer as an attribute to the widget

Reply:
1. We need better error message, indicating to the developer using widget what's going on
2. I would provide defaults at least in react example. But honestly I think we could also provide default rpc URLs for networks as well (user would override them if he provides them).

cc @MakMuftic @itsbobbyzzz168

Reply:
@mpetrunic for keeping defaults widget will require urls of substrate chains to create ApiPromise objects

Reply:
I know, but we know those

Reply:
@mpetrunic I don't think they are part of SDK or configuration